### PR TITLE
Liquibase - Record service classes implementation instead of  generating a native resource file for each service interface

### PR DESCRIPTION
**Pull request by machi1990**
This cleans up the extension processing part a little bit as we can record these info in a registry
and do a map lookup during runtime instead of reading the generated resource file for each service
class

Follows up this https://github.com/quarkusio/quarkus/pull/8065#issuecomment-603217470 comment.

### CODE REVIEW

File: extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java
Comment (gastaldi) [inline extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: Should we care about the order here (`LinkedHashMap`) ?

Comment (gastaldi) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: > Should we care about the order here (`LinkedHashMap`) ?

Probably yes, to avoid uncertainty while loading each service, WDYT?

Comment (machi1990) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: > > Should we care about the order here (`LinkedHashMap`) ?
> 
> Probably yes, to avoid uncertainty while loading all services, WDYT?

I do not think so. I think we just need a lookup table here, order is irrelevant. What uncertainty are you thinking of?

Comment (gastaldi) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: Ah nevermind, I thought you were iterating entries. I should stop reviewing code late at night :stuck_out_tongue:

**Review by gastaldi — APPROVED** [review]

**Comment by gastaldi** [comment]
I'll let @gsmet review and merge

**Comment by machi1990** [comment]
Thanks @gastaldi

**Comment by gsmet** [comment]
I rebased it, let's wait for CI.

**Review by gsmet — APPROVED** [review]

**Pull request merged**
